force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,189 @@
+public with sharing class OpportunityTriggerHandler extends TriggerHandler {
+    private static Boolean hasAlreadyRun = false;
+    private static Boolean hasAlreadyRunInsert = false; // Add this line
+
+    protected override void beforeInsert() {
+        setTypeForOpportunities((List<Opportunity>) Trigger.new);
+    }
+
+    protected override void beforeUpdate() {
+        validateOpportunityAmount((List<Opportunity>) Trigger.new);
+        setPrimaryContactCEO((List<Opportunity>) Trigger.new);
+    }
+
+    protected override void beforeDelete() {
+        preventClosedOpportunityDeletion((List<Opportunity>) Trigger.old);
+    }
+
+    protected override void afterInsert() {
+
+        if (Trigger.isInsert && Trigger.isAfter && !hasAlreadyRunInsert) { 
+            hasAlreadyRunInsert = true; // Add this line
+            createFollowUpTasks((List<Opportunity>) Trigger.new);
+        }
+    }
+
+    protected override void afterUpdate() {
+        if (hasAlreadyRun) return;
+        hasAlreadyRun = true;
+
+        updateOpportunityDescriptionOnStageChange((List<Opportunity>) Trigger.new, (Map<Id, Opportunity>) Trigger.oldMap);
+    }
+
+    protected override void afterDelete() {
+        notifyOpportunityDeletion((List<Opportunity>) Trigger.old);
+    }
+
+    protected override void afterUndelete() {
+        assignPrimaryContactVP((List<Opportunity>) Trigger.new);
+    }
+
+    
+
+    private void setTypeForOpportunities(List<Opportunity> opps) {
+        for (Opportunity opp : opps) {
+            if (opp.Type == null) {
+                opp.Type = 'New Customer';
+            }
+        }
+    }
+
+    private void validateOpportunityAmount(List<Opportunity> opps) {
+        for (Opportunity opp : opps) {
+            if (opp.Amount != null && opp.Amount < 5000) {
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+    }
+
+    private void setPrimaryContactCEO(List<Opportunity> opps) {
+        assignPrimaryContactByTitle(opps, 'CEO');
+    }
+
+    private void preventClosedOpportunityDeletion(List<Opportunity> opps) {
+        for (Opportunity opp : opps) {
+            if (opp.StageName == 'Closed Won' || opp.StageName == 'Closed Lost') {
```

Comment:
Since the direction for this ask for Banking and closed won, try to use the isWon field on opportunities.